nerdtree_plugin/file_filter.vim

```diff
@@ -28,36 +28,78 @@ if !exists('g:NERDTreeFileFilterEnabled')
     let g:NERDTreeFileFilterEnabled = 0
 endif
 
+function! NERDTreeDirectoryFilter(params)
+    if g:NERDTreeFileFilterEnabled != 1
+        return 0
+    endif
+
+    let path = a:params['path']
+
+    if path.isDirectory
+        let tree = a:params['nerdtree']
+        let node = b:NERDTreeRoot.findNode(path)
```

Comment:
I think this may be the main reason for performance issues. I guess this searches the whole tree from the root to find the matching node. It would be nice to search the `a:params['nerdtree']` param, but I couldn't easily find a way to do that.